recipes/up-cpp/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+from conan.tools.scm import Version, Git
+import os
+
+required_conan_version = ">=1.59.0"
+
+class UpCppConan(ConanFile):
+    name = "up-cpp"
+    description = "C++ language implementation of a uProtocol language SDK"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/eclipse-uprotocol/up-cpp"
+    topics = ("uprotocol", "sdv", "middleware")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    # TODO: Upon release of up-spec 1.5.8, this will need to be updated to:
+    # proto_containing_folder = "up-spec"
+    proto_containing_folder = "up-core-api"
+    # TODO: Upon release of up-spec 1.5.8, this will need to be updated to:
+    # proto_subfolder = "up-core-api"
+    proto_subfolder = "."
+    proto_folder = "uprotocol"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_testing": [True, False],
+        "build_unbundled": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "build_testing": False,
+        "build_unbundled": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    # unclear on what I should choose for msvc and Visual Studio
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "13",
+            "clang": "13",
+            "gcc": "11",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+        self.folders.build = os.path.join("build", str(self.settings.build_type))
+        self.folders.install = "install"
+
+    def build_requirements(self):
+        self.build_requires("cmake/[>=3.18.0]")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.12")
+        self.requires("spdlog/1.13.0")
+        if self.options.build_testing:
+            self.requires("gtest/1.14.0")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        if is_msvc(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} need to have paths in CMake files updated to handle both \ and /."
```

Comment:
What do you mean with this comment? Does up-cpp support Windows? In case is a bug related how Conan manages the path, you still can replace "\\" by "/", for instance.